apps/system/js/notifications.js

```diff
@@ -403,44 +402,57 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var cIndex = this.unReadNotificationIndex(manifestURL);
+    if (cIndex === -1) {
     // We need to animate the ambient indicator when the toast
     // timesout, so we skip updating it here, by passing a skip bool
-    this.addUnreadNotification(detail.id, true);
-
+    this.addUnreadNotification(manifestURL, true);
+    cIndex = this.unReadNotificationIndex(manifestURL);
     var notificationNode = document.createElement('div');
     notificationNode.classList.add('notification');
     notificationNode.setAttribute('role', 'link');
-
+    notificationNode.setAttribute('id', manifestURL);
+    notificationNode.setAttribute('tabindex', '0');
     notificationNode.dataset.notificationId = detail.id;
     notificationNode.dataset.noClear = behavior.noclear ? 'true' : 'false';
     notificationNode.lang = detail.lang;
     notificationNode.dataset.predefinedDir = detail.bidi;
-
     notificationNode.dataset.obsoleteAPI = 'false';
     if (typeof detail.id === 'string' &&
-        detail.id.indexOf('app-notif-') === 0) {
+      detail.id.indexOf('app-notif-') === 0) {
       notificationNode.dataset.obsoleteAPI = 'true';
     }
     var type = detail.type || 'desktop-notification';
     notificationNode.dataset.type = type;
     notificationNode.dataset.manifestURL = manifestURL;
-
-    if (detail.icon) {
+    if (detail.appIcon) {
       var icon = document.createElement('img');
-      icon.src = detail.icon;
+      icon.src = detail.appIcon;
       icon.setAttribute('role', 'presentation');
       notificationNode.appendChild(icon);
     }
-
+    var smsManifestUrl = 'app://sms.gaiamobile.org/manifest.webapp';
     var titleContainer = document.createElement('div');
     titleContainer.classList.add('title-container');
-
+    var countSpan = document.createElement('span');
+    countSpan.classList.add('detail-count');
+    countSpan.textContent = '';
+    titleContainer.appendChild(countSpan);
     var title = document.createElement('div');
     title.classList.add('title');
-    title.textContent = detail.title;
+    // in case of text message, title in 'detail' object is name of the sender
+    // but as per the UX title will be 'new messages'.
+    if(detail.manifestURL === smsManifestUrl) {
```

Comment:
you are making separate processing only for sms messages, but as per UI spec there at least 3 different types of 'titles' for notifications. Maybe it will be better to move customization of ``notificationNode`` according to type of application to the separate utility function?